cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -174,8 +194,16 @@ def index_item(item, skip_index=False):
         try:
             with modulestore.branch_setting(ModuleStoreEnum.RevisionOption.published_only):
                 structure = cls._fetch_top_level(modulestore, structure_key)
+                groups_usage_info = None
+                if cls.DOCUMENT_TYPE == "courseware_content":
+                    groups_usage_info = GroupConfiguration.get_content_groups_usage_info(modulestore, structure).items()
+                    if groups_usage_info:
+                        for name, group in groups_usage_info:  # pylint: disable=unused-variable
+                            for module in group:
+                                view, args, kwargs = resolve(module['url'])  # pylint: disable=unused-variable
+                                module['usage_key_string'] = kwargs['usage_key_string']
                 for item in structure.get_children():
-                    index_item(item)
+                    index_item(item, False, groups_usage_info)
```

Comment:
Perhaps omitting the 2nd argument in this call, and naming the 3rd one, since the 2nd parameter is set to `False` by default.